common/core/desktop/tests/unit/kmx/kmx.cpp

```diff
@@ -313,7 +326,7 @@ int run_test(const km::kbp::path & source, const km::kbp::path & compiled) {
 
   // Run through key events, applying output for each event
   for (auto p = next_key(keys); p.vk != 0; p = next_key(keys)) {
-    try_status(km_kbp_process_event(test_state, p.vk, p.modifier_state));
+    try_status(km_kbp_process_event(test_state, p.vk, p.modifier_state | caps_lock_state(), 1));
 
     for (auto act = km_kbp_state_action_items(test_state, nullptr); act->type != KM_KBP_IT_END; act++) {
       apply_action(test_state, *act, text_store, options);
```

Comment:
For the tests, we need to maintain the Caps Lock state as if it was the system doing it. So before calling km_kbp_process_event, we should be updating Caps Lock.

We also need to be passing in key up events to properly exercise the caps lock state changes, and then processing any actions arising from that as well.

```suggestion
    if(p.vk == KM_KBP_VKEY_CAPS) {
      toggle_caps_lock_state();
    }
    
    for(auto key_down = 1; key_down >= 0; key_down--) {
      try_status(km_kbp_process_event(test_state, p.vk, p.modifier_state | caps_lock_state(), key_down));
      for (auto act = km_kbp_state_action_items(test_state, nullptr); act->type != KM_KBP_IT_END; act++) {
        apply_action(test_state, *act, text_store, options);
      }
```

Reply:
done